shoop/admin/modules/suppliers/views/edit.py

```diff
@@ -29,6 +31,14 @@ class SupplierEditView(CreateOrUpdateView):
     template_name = "shoop/admin/suppliers/edit.jinja"
     context_object_name = "supplier"
 
+    def get_object(self, queryset=None):
+        obj = super(SupplierEditView, self).get_object(queryset)
+        if not settings.SHOOP_ENABLE_MULTIPLE_SUPPLIERS:
```

Comment:
Should this be refactored into a free function (along with the equivalent code for Shops) like

```python
def check_only_one_allowed(setting_name, obj):
    if getattr(settings, setting_name, True):
        return
    if not obj.pk and obj.__class__.objects.count() >= 1:
        raise Problem(_("Only one %(model)s permitted.") % {"model": obj._meta.verbose_name})
# ...
check_only_one_allowed("SHOOP_ENABLE_MULTIPLE_SUPPLIERS", obj)
# ...
check_only_one_allowed("SHOOP_ENABLE_MULTIPLE_SHOPS", obj)
```